File: LatihanInheritance/Mobil.cpp
#include "Kendaraan.hpp"

class Mobil : public Kendaraan {
private:
    string transportBrand;
    int releaseYear;
    string supir;

public:
    Mobil() : Kendaraan() {
        this->transportBrand = "XXX";
        this->releaseYear = 0;
        this->supir = "XXX";
    }

    Mobil(int transportID, string transportBrand, int releaseYear, string supir) : Kendaraan(transportID) {
        this->transportBrand = transportBrand;
        this->releaseYear = releaseYear;
        this->supir = supir;
    }

    Mobil(const Mobil &mobil) : Kendaraan(mobil) {
        this->transportBrand = mobil.transportBrand;
        this->releaseYear = mobil.releaseYear;
        this->supir = mobil.supir;
    }

    Mobil& operator= (const Mobil &mobil) {
        Kendaraan::operator=(mobil);
        this->transportBrand = mobil.transportBrand;
        this->releaseYear = mobil.releaseYear;
        this->supir = mobil.supir;
        return *this;
    }

    ~Mobil() {}

    void printInfo() {
        cout << "Berikut informasi tentang mobil ini: " << endl;
        cout << "ID Kendaraan: " << this->transportID << endl;
        cout << "Merk: " << this->transportBrand << endl;
        cout << "Tahun Keluaran: " << this->releaseYear << endl;
        cout << "Nama Supir: " << this->supir << endl;
    }

    int biayaSewa(int lamaSewa) {
        return 500000 * lamaSewa;
    }
};

File: LatihanInheritance-Java/Mobil.java
// Aria Judhistira / 13523112 / K02

public class Mobil extends Kendaraan {
    private String merk;
    private int tahunKeluaran;
    private String namaSupir;

    public Mobil() {
        super();
        this.merk = "XXX";
        this.tahunKeluaran = 0;
        this.namaSupir = "XXX";
    }

    public Mobil(int kendaraanID, String merk, int tahun, String nama) {
        super(kendaraanID);
        this.merk = merk;
        this.tahunKeluaran = tahun;
        this.namaSupir = nama;
    }

    public Mobil(Mobil mobil) {
        super(mobil);
        this.merk = mobil.merk;
        this.tahunKeluaran = mobil.tahunKeluaran;
        this.namaSupir = mobil.namaSupir;
    }

    @Override
    public void printInfo() {
        System.out.println("Berikut informasi mobil ini: ");
        System.out.println("ID Kendaraan: " + kendaraanID);
        System.out.println("Merk: " + merk);
        System.out.println("Tahun Keluaran: " + tahunKeluaran);
        System.out.println("Nama Supir: " + namaSupir);
    }

    @Override
    public int biayaSewa(int lamaSewa) {
        return 500000 * lamaSewa;
    }
    
}
